.github/workflows/build.yml

```diff
@@ -276,90 +258,230 @@ jobs:
+      zip_name: consul_${{ needs.get-product-version.outputs.product-version }}_linux_${{ matrix.arch }}.zip
+
+    name: Verify ${{ matrix.arch }} linux binary
     steps:
       - uses: actions/checkout@v2
-      - name: Download artifact
-        uses: actions/download-artifact@v2
+
+      - name: Download ${{ matrix.arch  }} zip
+        uses: actions/download-artifact@v3
         with:
-          name: ${{ env.PKG_NAME }}_${{ env.version }}_linux_${{ matrix.arch }}.zip
-      - name: Run verification for ${{ matrix.arch }} linux zip
+          name: ${{ env.zip_name }}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+        if: ${{ matrix.arch == 'arm' || matrix.arch == 'arm64' }}
+        with:
+          # this should be a comma-separated string as opposed to an array
+          platforms: arm,arm64
+
+      - name: Unzip ${{ matrix.arch }} zip
+        run: unzip ${{ env.zip_name }}
+
+      - name: Run verification for ${{ matrix.arch }} binary
+        if: ${{ matrix.arch == '386' || matrix.arch == 'amd64' }}
+        run: .github/scripts/verify_bin.sh ./consul v${{ env.version }}
+
+      - name: Run verification for ${{ matrix.arch }} binary (using Docker linux/arm/v7)
+        if: ${{ matrix.arch == 'arm' }}
+        run: |
+          docker run --platform=linux/arm/v7 -v $(pwd):/workdir -w /workdir  \
+            arm32v7/debian /workdir/.github/scripts/verify_bin.sh ./consul v${{ env.version }}
+
+      - name: Run verification for ${{ matrix.arch }} binary (using Docker linux/arm64)
+        if: ${{ matrix.arch == 'arm64' }}
         run: |
-          unzip ${{ env.PKG_NAME }}_${{env.version}}_linux_${{ matrix.arch }}.zip
-          ./consul version
+          docker run --platform=linux/arm64 -v $(pwd):/workdir -w /workdir  \
+            arm64v8/debian /workdir/.github/scripts/verify_bin.sh ./consul v${{ env.version }}
 
   verify-darwin:
     needs:
       - get-product-version
       - build-darwin
     runs-on: macos-latest
     strategy:
-      matrix:
-        arch: ["amd64", "arm64"]
       fail-fast: true
     env:
       version: ${{needs.get-product-version.outputs.product-version}}
-    name: Verify ${{ matrix.arch }} darwin zip
+      zip_name: consul_${{ needs.get-product-version.outputs.product-version }}_darwin_amd64.zip
+
+    name: Verify amd64 darwin binary
     steps:
       - uses: actions/checkout@v2
-      - name: Download artifact
-        uses: actions/download-artifact@v2
+
+      - name: Download amd64 darwin zip
+        uses: actions/download-artifact@v3
         with:
-          name: ${{ env.PKG_NAME }}_${{env.version}}_darwin_${{ matrix.arch }}.zip
-      - name: Run verification for ${{ matrix.arch }} linux zip
-        if: ${{ matrix.arch == 'amd64' }}
-        run: |
-          unzip ${{ env.PKG_NAME }}_${{env.version}}_darwin_${{ matrix.arch }}.zip
-          ./consul version
+          name: ${{ env.zip_name }}
+
+      - name: Unzip ${{ matrix.arch }} darwin zip
```

Comment:
Since verify-darwin workflow doesn't have the arch matrix, not sure if it will cause any issue in these steps.

Reply:
Good call. I'll update it to use a static string.